crates/tasks/src/lib.rs

```diff
@@ -404,12 +404,24 @@ impl TaskExecutor {
 impl TaskSpawner for TaskExecutor {
     fn spawn(&self, fut: BoxFuture<'static, ()>) -> JoinHandle<()> {
         self.metrics.inc_regular_task();
-        self.spawn(fut)
+        // Wrap the original future to increment the finished tasks counter upon completion
+        let metrics = self.metrics.clone();
+        let wrapped_fut = async move {
+            let _ = fut.await;
+            metrics.inc_finisehd_regular_task(); // Increment the number of finished tasks
+        };
+        self.spawn(wrapped_fut)
```

Comment:
I'd like to move this to:

https://github.com/paradigmxyz/reth/blob/1ec4d462a22574ddf3baa88e507d6fbbf879eb40/crates/tasks/src/lib.rs#L274-L278